bcipy/display/demo/components/demo_layouts.py

```diff
@@ -0,0 +1,288 @@
+"""Useful for viewing computed positions in a window"""
+import time
+from typing import Callable, List, Tuple, Union
+
+from psychopy import visual
+from psychopy.visual.circle import Circle
+from psychopy.visual.line import Line
+from psychopy.visual.rect import Rect
+from psychopy.visual.shape import ShapeStim
+
+from bcipy.display.components.layout import (Layout, at_top, centered,
+                                             envelope, height_units,
+                                             scaled_size)
+from bcipy.display.paradigm.vep.layout import BoxConfiguration, checkerboard
+
+
+def make_window():
+    """Make a sample window on which to draw."""
+    return visual.Window(size=[700, 500],
+                         fullscr=False,
+                         screen=0,
+                         waitBlanking=False,
+                         color='black',
+                         winType='pyglet')
+
+
+def draw_boundary(win,
+                  layout: Layout,
+                  color: str = 'blue',
+                  line_px: int = 5,
+                  name: str = None):
+    """Display the layout's outline."""
+    if name:
+        print(f"Drawing boundary for {name}")
+    top = Line(win=win,
+               units=layout.units,
+               start=(layout.left, layout.top),
+               end=(layout.right, layout.top),
+               lineColor=color,
+               lineWidth=line_px)
+
+    bottom = Line(win=win,
+                  units=layout.units,
+                  start=(layout.left, layout.bottom),
+                  end=(layout.right, layout.bottom),
+                  lineColor=color,
+                  lineWidth=line_px)
+
+    left = Line(win=win,
+                units=layout.units,
+                start=(layout.left, layout.top),
+                end=(layout.left, layout.bottom),
+                lineColor=color,
+                lineWidth=line_px)
+    right = Line(win=win,
+                 units=layout.units,
+                 start=(layout.right, layout.top),
+                 end=(layout.right, layout.bottom),
+                 lineColor=color,
+                 lineWidth=line_px)
+    border = [top, right, bottom, left]
+    for line in border:
+        print(f"Drawing line from {line.start} to {line.end}")
+        line.draw()
+
+
+def draw_position(win: visual.Window,
+                  pos: Tuple[float, float],
+                  color: str = 'blue',
+                  size: Union[float, Tuple[float, float]] = 0.025,
+                  name: str = None):
+    """Draw the provided positions"""
+    if name:
+        print(f"Drawing position {name} at {pos}")
+    circle = Circle(win, pos=pos, color=color, size=size)
+    circle.draw()
+
+
+def draw_positions(win: visual.Window,
```

Comment:
It seems like we decide the boundaries of the boxes here. Is there any way to extract the coordinates of individual symbols in a box?

Reply:
This demo code is useful for visualizing a list of positions to se where they are at in the window. For example, I used this to confirm the checkerboard positions and layouts. It could also be used for matrix symbol positions. But to extract coordinates within a box I think you would need to add some interactivity.

I think it's probably better, though, to compute positions whenever possible. The layout module is useful for this. For instance, I could create a Layout based on the box corners and used its functions to compute the center, etc.